### PR TITLE
[tripy] Sequential module

### PR DESCRIPTION
**Pull request by farazkh80**
Sequential similar to [torch.nn.sequential ](https://pytorch.org/docs/stable/generated/torch.nn.Sequential.html)

Things done

- [x] new frontend module called `tp.Sequential`
- [x] Unit tests under `tests/frontend/module/test_sequential.py`
- [x] integration tests and comparison with torch under

Supports nested tp.Sequential and most list operations other than modifications. `tp.Sequential` can not be modified after creation.

### CODE REVIEW

File: tripy/tripy/frontend/module/sequential.py [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+
+            model = tp.Sequential(tp.Linear(1, 3), tp.Linear(3, 2))
+            for layer in model:
+                print(layer)
+        """
+        return iter(self._modules.values())
+
+    def __getitem__(self, idx: Union[int, str]) -> Module:
+        r"""
+        Accesses a module by index (int) or name (str).
```
Comment (pranavm-nvidia) [inline tripy/tripy/frontend/module/sequential.py:100]: Is `int` here valid syntax? I think the interpreter would reject something like `module.0`.

Comment (slyubomirsky) [reply tripy/tripy/frontend/module/sequential.py:100]: I think it would be fine. `__getitem__` is array-style access.

Comment (pranavm-nvidia) [reply tripy/tripy/frontend/module/sequential.py:100]: Ah my mistake, I misread it as `__getattr__`.

**Comment by farazkh80** [comment]
closing sin e I have a feature branch MR now

**Pull request closed without merging**